reference-implementation/lib/writable-stream.js

```diff
@@ -551,11 +535,8 @@ function WritableStreamDefaultWriterClose(writer) {
     return Promise.reject(new TypeError(
       `The stream (in ${state} state) is not in the writable state and cannot be closed`));
   }
-  if (stream._pendingAbortRequest !== undefined) {
-    return Promise.reject(new TypeError('Requested to abort'));
-  }
 
-  assert(state === 'writable');
+  assert(state === 'writable' || state === 'erroring');
```

Comment:
writer.close() is useful to wait for the sink to become quiescent. You can also do this with writer.closed, but that would mean writing code like

```javascript
return writer.close().catch(() => writer.closed);
```

Also I want this to just work without the developer having to think about it. If there is a developer error like calling close() twice on the same stream then close() will reject immediately, but otherwise it consistently waits for no underlying sink methods to be running.

write() never provides any such guarantee, so rejecting immediately when we know there is no hope of the write() working is the best behaviour.